AWSClientRuntime/Sources/Middlewares/Sha256TreeHashMiddleware.swift

```diff
@@ -0,0 +1,118 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0.
+
+import AwsCommonRuntimeKit
+import ClientRuntime
+import AwsCCal
+
+public struct Sha256TreeHashMiddleware<OperationStackOutput: HttpResponseBinding,
+                                       OperationStackError: HttpResponseBinding>: Middleware {
+    public let id: String = "Sha256TreeHash"
+    
+    private let sha256TreeHashHeaderName = "X-Amz-Sha256-Tree-Hash"
+    
+    private let sha256HashHeaderName = "X-Amz-Content-Sha256"
```

Comment:
I think the code would read better if the variables were renamed to:
X_AMZ_SHA256_TREE_HASH_HEADER_NAME
X_AMZ_CONTENT_SHA256_HEADER_NAME